### ISSUE TITLE
Watch test failing with "TypeError: Cannot read property 'close' of undefined"

### ISSUE THREAD

**Issue by fox1t** [fastify/fastify-cli#271]
## 🐛 Bug Report

The test ["should start the server with watch options that the child process restart when directory changed" ](https://github.com/fastify/fastify-cli/blob/master/test/start.test.js#L527), __that is skipped on GitHub Actions__, on local machine trhows uncaught TypeError :
```
test/start.test.js 1> TypeError: Cannot read property 'close' of undefined
test/start.test.js 1> [fastify-cli] app crashed - waiting for file changes before starting...
```
The problem is [line 23](https://github.com/fastify/fastify-cli/blob/master/lib/watch/fork.js#L23) of `fork.js`. When the process restarts because of [line 88](https://github.com/fastify/fastify-cli/blob/master/lib/watch/index.js#L88) of `watch/index.js` `fastify` variable is still `undefined`.

Checking if fastify is not `undefined` fixes the issue and shows `test/start.test.js 1> [fastify-cli] process forced end` message.

```js
process.on('message', function (event) {
  if (event === GRACEFUL_SHUT) {
    const message = chalk.red('[fastify-cli] process forced end')
    setTimeout(exit.bind({ message }), TIMEOUT).unref()
    if(fastify) {
      fastify.close(() => {
        process.exit(0)
      })
    }
  }
})
```

## To Reproduce

Run the test suite.

## Expected behavior

Restart of the process without TypeError.

## Your Environment

- *node version*: 14.7.0
- *os*: Mac

**Comment by fox1t** [comment]
ping @mcollina

**Comment by mcollina** [comment]
It passes locally!

**Comment by fox1t** [comment]
Yes, it passes because [this](https://github.com/fastify/fastify-cli/blob/master/lib/watch/fork.js#L30) catches it, but the TypeError is present. Or are you saying that you don't have 
```
test/start.test.js 1> TypeError: Cannot read property 'close' of undefined
test/start.test.js 1> [fastify-cli] app crashed - waiting for file changes before starting...
```
printed in your console?

**Comment by alemagio** [comment]
I have the same error in console while running tests.
But the tests goes on without failing.

**Comment by fox1t** [comment]
Nice! It can't failed because of that uncaughtException listener that just exists the child process. The related tests are checking if the process exits and not why it exists. It should exit because there was a restart beacuse of changed files, but instead it exists because of a TypeError.

**Comment by alemagio** [comment]
Would you mind if I try fixing it?

**Comment by fox1t** [comment]
Not at all! You can find the fix on the snippet in this issue. However, I want to have this bug confirmed by @mcollina or some other core team member before fixing on it.

**Comment by mcollina** [comment]
Yes this is confirmed.

**Comment by fox1t** [comment]
@alemagio go for the fix then! After it gets merge I'll publish a new version since we have 2 PR pending publications.

**Comment by alemagio** [comment]
>> By the way, why that test is skipped on GitHub Actions?

The test is skipped on github actions.
Should I remove the `{ skip: onGithubAction }`?

**Comment by fox1t** [comment]
I don't know why we should skip it on GitHub. The only thing that comes to my mind is because of process forks and watching files. Let's try to push it without skip and see how Actions will handle it! XD

**Comment by nooreldeensalah** [comment]
Is this still an issue? Or was this fixed by #272?
